src/ElvClient.js

```diff
@@ -1054,6 +1056,18 @@ class ElvClient {
     );
   }
 
+  /**
+   * Record the write token with node url to the HttpClient
+   *
+   * @param {string} writeToken - Write token to be cached
+   * @param {string=} nodeUrlStr - If provided, this node url
+   * will be associated with the provided write token, which will
+   * be used in subsequent HTTP calls
+   */
+  RecordWriteToken(writeToken, nodeUrlStr) {
+    this.HttpClient.RecordWriteToken(writeToken, nodeUrlStr);
+  }
```

Comment:
Should `nodeUrlStr` be optional here? I think `ElvClient.RecordWriteToken()` would only be called with `nodeUrlStr` always supplied.

Adding validation of inputs here would be helpful too  (e.g. `nodeURLValidateForWriteToken()` from sample code)

On further thought even `HttpClient.RecordWriteToken()` should probably be changed to make `nodeUrlStr` required - I think the auto-magic behavior makes things harder to understand/debug and I suspect it is also not bulletproof from a concurrency standpoint (if failover occurs, `HttpClient.uriIndex` could change in between call that generates write token and call to record it)

(Would need to replicate changes from https://github.com/eluv-io/elv-client-js/pull/192/files#diff-117e6126f26bd127ea27c77589f1e0d703d77d96e8dc6076372af8cf3cb493efR777-R790 everywhere that `HttpClient.RecordWriteToken()` is called without `nodeUrlStr`, I see two calls in ContentManagement.js, line 190 and line 618)

Reply:
Also update `UseRegion()` ?